### PR TITLE
PLATFORM-4222 | upgrade to PHP 7.3.7 on dev

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/PLATFORM-4222
https://wikia-inc.atlassian.net/browse/CORE-260

https://www.php.net/releases/7_3_7.php

```
$ docker run -it --rm artifactory.wikia-inc.com/platform/php-wikia-devbox:latest -v
PHP 7.3.7 (fpm-fcgi) (built: Jul 17 2019 22:34:16)
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.7, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.7, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.7.2, Copyright (c) 2002-2019, by Derick Rethans
```

### This pull request updates

* PHP to v7.3.7
* PHP extensions in base image: mustache to v0.8.1 and tideways to v4.1.7
* PHP extensions for dev environment: uopz to v6.01, xdebug to v2.7.2 and yaml to v2.0.4

### Additional changes

* PHP 7.3 has fixed handling of logging to stdout from fpm workers. We got rid of a #15173 hack involving a small container called `mediawiki-logger` + MediaWiki logging via socket

### Performance

PHP 7.3 is ~12% faster than 7.0 (median response time for ~1.5k various URLs).

### CODE REVIEW

File: includes/OutputPage.php [hunk 1]
```diff
@@ -2745,6 +2745,8 @@ public function getResourceLoader() {
 	public function makeResourceLoaderLink( $modules, $only, $useESI = false, array $extraQuery = array(), $loadCall = false ) {
 		global $wgResourceLoaderUseESI;
 
+		$modules = (array) $modules; // Wikia change - PHP 7.3
```
Comment (pwojcik86) [inline includes/OutputPage.php:2746]: Can `$modules` be declared as an `array` instead of this cast?

Comment (macbre) [reply includes/OutputPage.php:2746]: I've simply "backported" the approach from MediaWiki's upstream - https://github.com/wikimedia/mediawiki/blob/c4e284f1130cdc2a4811120594bdb8013d17eaaa/includes/OutputPage.php#L3303

But sure, let's try adding a type hint in method declaration.

Comment (pwojcik86) [reply includes/OutputPage.php:2746]: I just read the comment for this method, and `$modules` is an `array` or `string`, so this is the only choice.

File: .travis.yml [hunk 2]
```diff
@@ -1,24 +1,26 @@
 language: php
-php: 7.0
-dist: trusty
+php: 7.3
+dist: xenial
```
Comment (mszabo-wikia) [inline .travis.yml:3]: 👍

**Review by mszabo-wikia — APPROVED** [review]

**Comment by themech** [comment]
taking it over

File: docker/base/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # This is a base Docker image used by Wikia's MediaWiki app. It uses base PHP image provided by Docker with additional
 # production dependencies installed.
-FROM php:7.0.30-fpm-stretch
+FROM php:7.3.3-fpm-stretch
```
Comment (mszabo-wikia) [inline docker/base/Dockerfile]: 7.3.7 on buster is out

Comment (themech) [reply docker/base/Dockerfile]: yeah, done in https://github.com/Wikia/app/pull/16873/commits/7f3999d4b05bfeccc44115b6d1bc1daaa9328de7

**Review by mszabo-wikia — APPROVED** [review]

**Comment by macbre** [comment]
:+1:

**Review by harnash — APPROVED** [review]

**Review by mszabo-wikia — APPROVED** [review]

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]
It seems we're missing some more backports (listed on https://phabricator.wikimedia.org/T206988):
* https://gerrit.wikimedia.org/r/c/mediawiki/core/+/45485 — warning in `LanguageConverter`
* https://gerrit.wikimedia.org/r/c/mediawiki/extensions/ParserFunctions/+/467306 — warning in `ExprParser`

**Comment by themech** [comment]
Good point @mszabo-wikia. I've ported the changes from upstream in the last 4 commits.

**Review by Grunny — APPROVED** [review]

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**